js/views/create_group_dialog_view.js

```diff
@@ -179,18 +179,39 @@
       this.$el.append(this.dialogView.el);
       return this;
     },
-    onSubmit(newMembers) {
+    async onSubmit(newMembers) {
       const ourPK = textsecure.storage.user.getNumber();
       const allMembers = window.Lodash.concat(newMembers, [ourPK]);
 
       // We need to NOT trigger an group update if the list of member is the same.
       const notPresentInOld = allMembers.filter(
         m => !this.existingMembers.includes(m)
       );
+
       const notPresentInNew = this.existingMembers.filter(
         m => !allMembers.includes(m)
       );
-      // would be easer with _.xor but for some reason we do not have it
+
+      // Filter out all linked devices for cases in which one device
+      // exists in group, but hasn't yet synced with its other devices.
+      const getDevicesForRemoved = async () => {
+        const promises = notPresentInNew.map(member =>
+          libloki.storage.getPairedDevicesFor(member)
+        );
+        const devices = window.Lodash.flatten(await Promise.all(promises));
+
+        return devices;
+      };
+
+      // Get all devices for notPresentInNew
+      const allDevicesOfMembersToRemove = await getDevicesForRemoved();
+
+      // If any extra devices of removed exist in newMembers, ensure that you filter them
+      const filteredMemberes = allMembers.filter(
+        member => !window.Lodash.includes(allDevicesOfMembersToRemove, member)
+      );
```

Comment:
don't get that part

Reply:
Forcibly removes slave devices from member list if it's a secondary device AND its primary is in the chat. Solves various quirky bugs